marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/Options.java

```diff
@@ -27,6 +27,7 @@
     private String initializeWriteApiPath;
     private String finalizeWriteApiPath;
     private JsonNode additionalExternalMetadata;
+    private String initializeReadApiPath;
```

Comment:
"readRowsEndpointParams" should be here as well, and it needs to be tested too. A good way to test that, along with the custom initialize endpoint, is the following:

- Don't load any customers, nor the TDE
- Have the initialize endpoint return a single static partition, regardless of what the other inputs are
- Have the initialize endpoint return a static Spark schema as well, regardless of what the other inputs are
- For readrowsendpointparams/endpointState, define a single JSON row that can be returned by the readRows endpoint
- Then have the readRows endpoint return whatever value is in endpointState (it'll also need to use something in the endpointState object to know whether it's been called, since you only want to call it once)

That will allow you to verify that both your custom initialize and read endpoints are being used.